### ISSUE TITLE
Edit Book crash at the end of text.

### ISSUE THREAD

**Issue by TomCatFort** [anegostudios/VintageStory-Issues#2572]
### Game Version

v1.18.0-rc.5

### Platform

Linux

### Modded

Vanilla

### SP/MP

Singleplayer

### Description

On the Edit Book UI, if I try to move the cursor to the right using the right arrow button, and the cursor is already at the end of the text, the game crashes.
It can happen with both the book title, and the body text UI fields.
It does not matter if the UI field is empty or it has some content.

### How to reproduce

**Repro in creative mode:**
1. Create a new book.
2. Begun editing the book using the Ink And Quill.
3. Select either the title, or the body to be edited.
4. (Optional) Enter some text, but make sure that the cursor is at the very end of the text when you done with this step.
5. Press the right arrow button on the keyboard.

### Screenshots

![book_edit_crash](https://user-images.githubusercontent.com/784076/231214105-f2ca1916-a4e1-4e2f-8373-08346217a3a9.png)

### Logs

**client-crash.txt**
```
Game Version: v1.18.0-rc.5 (Unstable)
4/11/2023 5:11:06 PM: Critical error occurred
Loaded Mods: game@1.18.0-rc.5, creative@1.18.0-rc.5, survival@1.18.0-rc.5
System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
  at System.Collections.Generic.List`1[T].get_Item (System.Int32 index) [0x00009] in <162bccb5b7ae4c18b9c0cff8fdc3dfbc>:0 
  at Vintagestory.GameContent.GuiDialogEditableBook.onTryTextChange (System.Collections.Generic.List`1[T] lines) [0x00008] in <6cfd4fb263c14bb2bb26b8c3db5dd864>:0 
  at Vintagestory.API.Client.GuiElementEditableTextBase.LoadValue (System.Collections.Generic.List`1[T] newLines) [0x0000a] in <d4680078b5a34c43896a1c31a64532d8>:0 
  at Vintagestory.API.Client.GuiElementEditableTextBase.SetValue (System.String text, System.Boolean setCaretPosToEnd) [0x00008] in <d4680078b5a34c43896a1c31a64532d8>:0 
  at Vintagestory.GameContent.GuiDialogReadonlyBook.updatePage (System.Boolean setCaretPosToEnd) [0x00067] in <6cfd4fb263c14bb2bb26b8c3db5dd864>:0 
  at Vintagestory.GameContent.GuiDialogEditableBook.OnKeyDown (Vintagestory.API.Client.KeyEvent args) [0x000f8] in <6cfd4fb263c14bb2bb26b8c3db5dd864>:0 
  at Vintagestory.Client.NoObf.GuiManager.OnKeyDown (Vintagestory.API.Client.KeyEvent args) [0x0009a] in <f54134278eac4e788d74f4fdeba7c0a6>:0 
  at Vintagestory.Client.NoObf.ClientMain.OnKeyDown (Vintagestory.API.Client.KeyEvent args) [0x0007f] in <f54134278eac4e788d74f4fdeba7c0a6>:0 
  at Vintagestory.Client.GuiScreenRunningGame.OnKeyDown (Vintagestory.API.Client.KeyEvent args) [0x00000] in <f54134278eac4e788d74f4fdeba7c0a6>:0 
  at Vintagestory.Client.ScreenManager.OnKeyDown (Vintagestory.API.Client.KeyEvent e) [0x0006e] in <f54134278eac4e788d74f4fdeba7c0a6>:0 
  at Vintagestory.Client.NoObf.ClientPlatformWindows.game_KeyDown (System.Object sender, OpenTK.Input.KeyboardKeyEventArgs e) [0x00082] in <f54134278eac4e788d74f4fdeba7c0a6>:0 
  at (wrapper delegate-invoke) System.EventHandler`1[OpenTK.Input.KeyboardKeyEventArgs].invoke_void_object_TEventArgs(object,OpenTK.Input.KeyboardKeyEventArgs)
  at OpenTK.NativeWindow.OnKeyDown (OpenTK.Input.KeyboardKeyEventArgs e) [0x00000] in <a4886e33f9fc416195d24950e7205b94>:0 
  at OpenTK.NativeWindow.OnKeyDownInternal (System.Object sender, OpenTK.Input.KeyboardKeyEventArgs e) [0x00000] in <a4886e33f9fc416195d24950e7205b94>:0 
  at (wrapper delegate-invoke) System.EventHandler`1[OpenTK.Input.KeyboardKeyEventArgs].invoke_void_object_TEventArgs(object,OpenTK.Input.KeyboardKeyEventArgs)
  at OpenTK.Platform.NativeWindowBase.OnKeyDown (OpenTK.Input.Key key, System.Boolean repeat) [0x0002e] in <a4886e33f9fc416195d24950e7205b94>:0 
  at OpenTK.Platform.X11.X11GLNative.ProcessEvents () [0x00713] in <a4886e33f9fc416195d24950e7205b94>:0 
  at OpenTK.NativeWindow.ProcessEvents (System.Boolean retainEvents) [0x00035] in <a4886e33f9fc416195d24950e7205b94>:0 
  at OpenTK.NativeWindow.ProcessEvents () [0x00000] in <a4886e33f9fc416195d24950e7205b94>:0 
  at OpenTK.GameWindow.Run (System.Double updates_per_second, System.Double frames_per_second) [0x000da] in <a4886e33f9fc416195d24950e7205b94>:0 
  at OpenTK.GameWindow.Run () [0x00000] in <a4886e33f9fc416195d24950e7205b94>:0 
  at Vintagestory.Client.ClientProgram.Start (Vintagestory.Client.ClientProgramArgs args, System.String[] rawArgs) [0x00421] in <9a07de3147e34b5c92f6fe3c6b341e52>:0 
  at Vintagestory.Client.ClientProgram+<>c__DisplayClass16_0.<.ctor>b__0 () [0x00000] in <9a07de3147e34b5c92f6fe3c6b341e52>:0 
  at Vintagestory.ClientNative.CrashReporter.Start (System.Threading.ThreadStart start) [0x00007] in <f54134278eac4e788d74f4fdeba7c0a6>:0 
```

**Comment by radfast** [comment]
Similar crash report with line numbers:

    System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
       at System.ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument argument, ExceptionResource resource)
       at Vintagestory.GameContent.GuiDialogEditableBook.onTryTextChange(List`1 lines) in VSSurvivalMod\Systems\WritingSystem\GuiDialogEditableBook.cs:line 78
       at Vintagestory.API.Client.GuiElementEditableTextBase.LoadValue(List`1 newLines) in VintagestoryApi\Client\UI\Elements\Impl\Interactive\Text\GuiElementEditableTextBase.cs:line 281
       at Vintagestory.API.Client.GuiElementEditableTextBase.SetValue(String text, Boolean setCaretPosToEnd) in VintagestoryApi\Client\UI\Elements\Impl\Interactive\Text\GuiElementEditableTextBase.cs:line 266
       at Vintagestory.GameContent.GuiDialogReadonlyBook.updatePage(Boolean setCaretPosToEnd) in VSSurvivalMod\Systems\WritingSystem\GuiDialogReadonlyBook.cs:line 174
       at Vintagestory.GameContent.GuiDialogEditableBook.OnKeyDown(KeyEvent args) in VSSurvivalMod\Systems\WritingSystem\GuiDialogEditableBook.cs:line 239
       at Vintagestory.Client.NoObf.GuiManager.OnKeyDown(KeyEvent args) in VintagestoryLib\Client\Systems\Gui\GuiManager.cs:line 361
       at Vintagestory.Client.NoObf.ClientMain.OnKeyDown(KeyEvent args) in VintagestoryLib\Client\ClientMain.cs:line 1736

**Comment by radfast** [comment]
Fixed in upcoming 1.18.2 stable, thank you for reporting
